tests/testthat/test-derive_locf_records.R

```diff
@@ -298,6 +298,134 @@ test_that("derive_locf_records Test 6: populate VISITNUM for LOCF records", {
   )
 
 
+  expect_dfs_equal(
+    base = expected_output,
+    compare = actual_output,
+    keys = c("STUDYID", "USUBJID", "PARAMCD", "PARAM", "AVISITN", "AVISIT", "DTYPE")
+  )
+})
+
+
+## Test 7: analysis visits are assigned based on time windows ----
+test_that("derive_locf_records Test 7: analysis visits are assigned based on time windows", {
+  input <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~PARAMCD, ~PARAM, ~AVALC, ~AVISITN, ~AVISIT, ~VISITNUM,
+    "TEST01", "01-701-1028", "DIABP", "Diastolic Blood Pressure (mmHg)", "79", 0, "BASELINE", 0,
+    "TEST01", "01-701-1028", "SYSBP", "Systolic Blood Pressure (mmHg)", "130", 0, "BASELINE", 0
+  )
+
+  input <- tibble::tribble(
```

Comment:
do we need both of these input datasets?  i think the second one overrides the first one

Reply:
Duplicate input datasets are removed